src/Smearing.jl

```diff
@@ -75,14 +71,32 @@ function occupation_derivative(S::FermiDirac, x)
         -exp(x) / (1+exp(x))^2
     end
 end
-
 # entropy(f) = -(f log f + (1-f)log(1-f)), where f = 1/(1+exp(x))
 # this "simplifies" to -(x*exp(x)/(1+exp(x)) - log(1+exp(x)))
 # although that is not especially useful...
+function xlogx(x)
+    iszero(x) ? zero(x) : x * log(x)
+end
 function entropy(S::FermiDirac, x)
     f = occupation(S, x)
     - (xlogx(f) + xlogx(1 - f))
 end
+function occupation_divided_difference(S::FermiDirac, x, y, εF, temperature)
+    temperature == 0 && return occupation_divided_difference(None(), x, y, εF, temperature)
```

Comment:
I think you should abbreviate temperature by `temp` or something like that. That word destroys the flow of reading in this function.

Reply:
Good point, done